leopiresgoss/tour-hunter-api
# Tour Hunter API

## TOUR HUNTER API

We implement an API for the project: Tour Hunter, where:

- An admin role can: Create or Delete tours.
- A client role can: Book a tour on the available dates.
- When a tour is created, the admin can add many images per tour.
- The images are stored in Cloudinary.
- The tour list endpoint doesn't need authentication.
- Other endpoints like: Create or Delete tours, and Book tours need user authentication.
- When a user sign up, the server send an email to confirm the account.
- When a user signs in, he receives a token that needs to be sent in the header for the endpoint that needs authentication.
- The documentation was elaborated using rsawg gem.

## Link to the API documentation
[Api Documentation](https://tourhunterapi.herokuapp.com/api-docs/index.html)

## Front-end Repository

You can find the Front-end repo [here](https://github.com/leopiresgoss/tour-hunter).

## PR of the Front-end project

You can find the PR of the Front-end project [here](https://github.com/leopiresgoss/tour-hunter/pull/31).

## Deployment

You can find the app live [here](https://tourhunterapp.herokuapp.com)

## Testing Locally
To test locally, you should:
- Clone the repo: `git@github.com:leopiresgoss/tour-hunter-api.git`
- Run: `bundle install`
- Create a `cloudinary.yml` file in the `config` folder
- Provide the credentials of your Cloudinary account
